main/guides/js-programming/agoric-js-overview.md

```diff
@@ -179,10 +176,7 @@ containing either of the strings `<!--` or `-->`, even if neither marks a commen
 
 ### Dynamic import expressions
 
-One active JS feature proposal would add a "dynamic import" expression: `await import('path')`. 
-If implemented (or if someone decides to be an early adopter and adds it to an engine), 
-and your JS engine has this, vat code might be able to bypass the `Compartment`'s module 
-map. For safety, the SES shim already rejects code that looks like it uses this feature. 
+The "dynamic import" feature (`await import('path')`) is not a function call but JavaScript syntax. As such it would allow vat code to bypass the shim's `Compartment`'s module map. For safety, the SES shim rejects code that looks like it uses this feature. 
```

Comment:
I see that both the old and new says `await import('path')` rather than just `import('path')`. This is wrong. Instead just say that the dynamic import expression returns a promise. If you think it would be helpful, show the longer expression below as a suggested usage pattern.

I would say "dynamic import expression" rather than "feature". The fact that it is an expression is important, to more clearly distinguish it from import declarations.

Reply:
I wanted to highlight that the bypass risk is due to the syntax nature of dynamic import. The fact that it can be used as an expression matters in that it makes it different in usage from static import declarations. If dynamic `import` was not a syntax but a function call, we would be able to trap them without compartments, and they wouldn't be a hazard.

I'll try to rewrite to make both dynamic expression and syntax more clear.

Reply:
Please change "would allow vat code to bypass" to "would let vat code bypass".

Reply:
@tyg done
@erights, I tried to address your concern while still highlighting the syntax aspect. PTAL